controller/src/test/java/io/pravega/controller/rest/v1/HealthTests.java

```diff
@@ -0,0 +1,502 @@
+ *
+ * Both implicit and explicit requests have the same code path(s) so we can expect results to be similar between each
+ * class of tests.
+ */
+@Slf4j
+public class HealthTests {
+
+    private static final String HOST = "localhost";
+
+    private static final String IMPLICIT_INDICATOR = "implicit-health-child";
+
+    private static final String IMPLICIT_COMPONENT = "health-component";
+
+    //Ensure each test completes within 30 seconds.
+    @Rule
+    public final Timeout globalTimeout = new Timeout(10000, TimeUnit.SECONDS);
+
+    private  HealthServiceFactory healthServiceFactory;
+
+    private RESTServerConfig serverConfig;
+    private RESTServer restServer;
+    private Client client;
+    private ConnectionFactory connectionFactory;
+    private HealthService service;
+
+    @Before
+    public void setup() throws Exception {
+        ControllerService mockControllerService = mock(ControllerService.class);
+        serverConfig = getServerConfig();
+        connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        healthServiceFactory = new HealthServiceFactory(Optional.ofNullable(getHealthConfig()));
+        service = healthServiceFactory.createHealthService(true);
+        restServer = new RESTServer(null, mockControllerService, null, serverConfig,
+                connectionFactory, service);
+        restServer.startAsync();
+        restServer.awaitRunning();
+        client = createJerseyClient();
+        // Must provide a HealthIndicator to the HealthComponent(s) defined by the HealthConfig.
+        service.registry().register(new StaticHealthyIndicator(IMPLICIT_INDICATOR), IMPLICIT_COMPONENT);
+    }
+
+    @After
+    public void teardown() {
+        service.clear();
+        // The ROOT component + the component supplied by our HealthConfig (getHealthConfig).
+        Assert.assertEquals(2, service.registry().contributors().size());
+        Assert.assertEquals(2, service.registry().components().size());
+        // Assert that the component does not contain children.
+        Assert.assertEquals("The HealthService should report only one component.",
+                1,
+                service.registry().get().contributors().size());
+    }
+
+    protected Client createJerseyClient() throws Exception {
+        return ClientBuilder.newClient();
+    }
+
+    RESTServerConfig getServerConfig() throws Exception {
+        return RESTServerConfigImpl.builder().host(HOST).port(TestUtils.getAvailableListenPort())
+                .build();
+    }
+
+    HealthConfig getHealthConfig() throws Exception {
+        return HealthConfigImpl.builder()
+                .define("health-component", StatusAggregatorImpl.DEFAULT)
+                .build();
+    }
+
+    String getURLScheme() {
+        return "http";
+    }
+
+        @After
+    public void tearDown() {
```

Comment:
Why do you have 2 teardown methods????

Reply:
This lets the JVM know that its *really* important to execute these teardown functions, making sure it won't forget. Lol, I based the file off `PingTest` and just overlooked removing it.